lib/qbwc/job.rb

```diff
@@ -9,6 +9,7 @@ def initialize(name, enabled, company, worker_class, requests = [])
     @worker_class = worker_class
     @requests = requests
     @request_index = 0
+    @requests_provided_when_job_added = get_persistent_value(:requests_provided_when_job_added)
```

Comment:
Why the new method `get_persistent_value` instead of calling the `requests_provided_when_job_added` method you added?